amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/persistence/mapper/TableMetaMapper.java

```diff
@@ -424,4 +430,74 @@ List<ServerTableIdentifier> selectTableIdentifiersByCatalog(
         typeHandler = MapLong2StringConverter.class)
   })
   List<TableRuntimeMeta> selectTableRuntimeMetas();
+
+  @Select(
+      "<script>"
+          + "<bind name=\"isMySQL\" value=\"_databaseId == 'mysql'\" />"
+          + "<bind name=\"isPostgreSQL\" value=\"_databaseId == 'postgres'\" />"
+          + "<bind name=\"isDerby\" value=\"_databaseId == 'derby'\" />"
+          + "SELECT table_id, catalog_name, db_name, table_name, current_snapshot_id, "
+          + "current_change_snapshotId, last_optimized_snapshotId, last_optimized_change_snapshotId, "
+          + "last_major_optimizing_time, last_minor_optimizing_time, last_full_optimizing_time, optimizing_status_code, "
+          + "optimizing_status_start_time, optimizing_process_id, "
+          + "optimizer_group, table_config, pending_input FROM table_runtime "
+          + "/* Debug: ${_databaseId} */"
+          + "WHERE 1=1 "
+          + "<if test='optimizerGroup != null'> AND optimizer_group = #{optimizerGroup} </if> "
+          + "<if test='fuzzyDbName != null and isMySQL'> AND db_name like CONCAT('%', #{fuzzyDbName, jdbcType=VARCHAR}, '%') </if>"
+          + "<if test='fuzzyDbName != null and (isPostgreSQL or isDerby)'> AND db_name like '%' || #{fuzzyDbName, jdbcType=VARCHAR} || '%' </if>"
+          + "<if test='fuzzyTableName != null and isMySQL'> AND table_name like CONCAT('%', #{fuzzyTableName, jdbcType=VARCHAR}, '%') </if>"
+          + "<if test='fuzzyTableName != null and (isPostgreSQL or isDerby)'> AND table_name like '%' || #{fuzzyTableName, jdbcType=VARCHAR} || '%' </if>"
```

Comment:
In my opinion, is it sufficient if we only support prefix search for DB and Table?
Using like '%xxx' for matching cannot enjoy the benefits brought by indexing.
@zhoujinsong CC

Reply:
+1 for only support prefix search, this respects the current logic we support now. and will change this after we reach the consense

Reply:
@majin1102 cc